props/prop_base.h

```diff
@@ -296,15 +312,51 @@ class PropBase : CommandParser, Looper, protected SaberBase {
   } while(0);
 
     ONCEPERBLADE(ACTIVATE);
+    #ifdef SAVE_STATE
+    ResumePreset();
+    #else
     SetPreset(0, false);
+    #endif
     return;
 
    bad_blade:
     STDOUT.println("BAD BLADE");
 #ifdef ENABLE_AUDIO
     talkie.Say(talkie_error_in_15, 15);
     talkie.Say(talkie_blade_array_15, 15);
-#endif    
+#endif
+  }
+
+  void ResumePreset() {
+    FileReader f;
+    SaveStateFile savestate;
+    if (!savestate.Read("curstate.ini")) {
+      savestate.Read("curstate.tmp");
+    }
+    SetPreset(savestate.preset, false);
+    if (savestate.volume <= VOLUME) {
```

Comment:
This still doesn't work if:

curstate.ini has the volume set to VOLUME
the dynamic mixer volume is less than VOLUME

The code will assume that the savefile doesn't have a volume in it and not set the volume, so the volume will remain low.